python/kernmlops/kernmlops_benchmark/mongodb.py

```diff
@@ -0,0 +1,90 @@
+import subprocess
+from dataclasses import dataclass
+from typing import Literal, cast
+
+from data_schema import GraphEngine, demote
+from kernmlops_benchmark.benchmark import Benchmark, GenericBenchmarkConfig
+from kernmlops_benchmark.errors import (
+  BenchmarkNotInCollectionData,
+  BenchmarkNotRunningError,
+  BenchmarkRunningError,
+)
+from kernmlops_config import ConfigBase
+
+
+@dataclass(frozen=True)
+class BenchmarkConfig(ConfigBase):
+  
+  recordCount: int = 1000000
+  readProportion: float = 0.25
+  updateProportion: float =0.75
+
+
+class MongoDbBenchmark(Benchmark):
+
+    @classmethod
+    def name(cls) -> str:
+        return "mongodb"
+
+    @classmethod
+    def default_config(cls) -> ConfigBase:
+        return BenchmarkConfig()
+
+    @classmethod
+    def from_config(cls, config: ConfigBase) -> "Benchmark":
+        generic_config = cast(GenericBenchmarkConfig, getattr(config, "generic"))
+        gap_config = cast(BenchmarkConfig, getattr(config, cls.name()))
+        return MongoDbBenchmark(generic_config=generic_config, config=gap_config)
+
+    def __init__(self, *, generic_config: GenericBenchmarkConfig, config: BenchmarkConfig):
+        self.generic_config = generic_config
+        self.config = config
+        self.benchmark_dir = self.generic_config.get_benchmark_dir() / self.name()
+        self.process: subprocess.Popen | None = None
+
+    def is_configured(self) -> bool:
+        return True
+        # print(f'is_configured directory name: {self.benchmark_dir}')
+        return self.benchmark_dir.is_dir()
+
+    def setup(self) -> None:
+        if self.process is not None:
+            raise BenchmarkRunningError()
+        self.generic_config.generic_setup()
+
+    def run(self) -> None:
+        if self.process is not None:
+            raise BenchmarkRunningError()
+
+        bash_file_path = self.benchmark_dir / "run_mongodb.sh" # Add the path to your bash file here
+        print(bash_file_path)
+        self.process = subprocess.Popen(
+            [
+                "bash",
+                bash_file_path,
```

Comment:
please cast this `Path` type to a string before passing it, i.e. `str(bash_file_path)`